packages/graphql/src/translate/queryAST/cypher-generators/CypherAnnotationSubqueryGenerator.ts

```diff
@@ -0,0 +1,251 @@
+        });
+
+        return cypherStatement;
+    }
+
+    private getNestedFieldsSubquery(nestedFields: Field[] | undefined): Cypher.Clause | undefined {
+        const target = this.returnVariable;
+        if (!nestedFields) return undefined;
+        assertIsCypherNode(target);
+
+        const nestedContext = new QueryASTContext({
+            target,
+            env: this.context.env,
+            neo4jGraphQLContext: this.context.neo4jGraphQLContext,
+        });
+
+        const nodeProjectionSubqueries = nestedFields.flatMap((field) => {
+            return field.getSubqueries(nestedContext).map((sq) => wrapSubqueryInCall(sq, target));
+        });
+        return Cypher.concat(...nodeProjectionSubqueries);
+    }
+
+    private getProjectionExpression({
+        fields,
+        nestedFields,
+    }: {
+        fields: Record<string, string> | undefined;
+        nestedFields?: Field[];
+    }): Cypher.Expr {
+        let projection: Cypher.Expr = this.returnVariable;
+
+        // nested fields are presents only if the attribute is an object or an abstract type, and they produce a different projection
+        if (nestedFields) {
+            projection = this.getNestedFieldsProjectionMap(nestedFields, this.returnVariable);
+        } else if (fields) {
+            projection = this.getFieldsProjectionMap(fields, this.returnVariable);
+        }
+
+        const collectedProjection = Cypher.collect(projection);
+
+        if (!this.attribute.isList()) {
+            return Cypher.head(collectedProjection);
+        }
+        return collectedProjection;
+    }
+
+    private getProjectionExpressionForUnionPartials(unionPartials: CypherUnionAttributePartial[]) {
+        const caseClause = new Cypher.Case();
+
+        for (const partial of unionPartials) {
+            const projection = partial.getProjectionExpression(this.returnVariable);
+            const predicate = partial.getFilterPredicate(this.returnVariable);
+            caseClause.when(predicate).then(projection);
+        }
+        const collectedProjection = Cypher.collect(caseClause);
+
+        if (!this.attribute.isList()) {
+            return Cypher.head(collectedProjection);
+        }
+        return collectedProjection;
+    }
+
+    private getNestedFieldsProjectionMap(fields: Field[], fromVariable: Cypher.Variable): Cypher.MapProjection {
+        const projection = new Cypher.MapProjection(fromVariable);
+
+        for (const field of fields) {
+            const fieldProjection = field.getProjectionField(fromVariable);
+            projection.set(fieldProjection);
+        }
+
+        return projection;
+    }
+
+    private getFieldsProjectionMap(
+        fields: Record<string, string>,
+        fromVariable: Cypher.Variable
+    ): Cypher.MapProjection {
+        const projection = new Cypher.MapProjection(fromVariable);
+        for (const [alias, name] of Object.entries(fields)) {
+            if (alias === name) projection.set(alias);
```

Comment:
I believe this is duplicate logic around the codebase, for instance: the method "createAttributeProperty" of the class "AttributeField", maybe this bit can be unified around the codebase!